packages/client/hmi-client/src/components/workflow/scenario-templates/counterfactual/counterfactual-scenario.ts

```diff
@@ -0,0 +1,273 @@
+		const modelConfig = await getModelConfigurationById(this.modelConfigId);
+		const interventionPolicy = await getInterventionPolicyById(this.interventionPolicyId);
+		const defaultSimulateState: SimulateCiemssOperationState = SimulateOp.initState!();
+		const defaultCalibrationState: CalibrationOperationStateCiemss = CalibrateOp.initState!();
+
+		// Add Nodes:
+		const modelNode = wf.addNode(ModelOp, POSITION, SIZE);
+
+		const modelConfigNode = wf.addNode(ModelConfigOp, POSITION, SIZE);
+
+		const interventionNode = wf.addNode(InterventionOp, POSITION, SIZE);
+
+		const datasetNode = wf.addNode(DatasetOp, POSITION, SIZE);
+
+		const calibrateNode = wf.addNode(CalibrateOp, POSITION, SIZE);
+
+		const simulateBaseNode = wf.addNode(SimulateOp, POSITION, SIZE);
+
+		const simulateWithInterventionNode = wf.addNode(SimulateOp, POSITION, SIZE);
+
+		const compareDatasetNode = wf.addNode(CompareDatasetOp, POSITION, SIZE);
+
+		// Update States:
+		// Model
+		wf.updateNode(modelNode, {
+			state: {
+				modelId: this.modelId
+			},
+			output: {
+				value: [this.modelId]
+			}
+		});
+		// Model Config:
+		wf.updateNode(modelConfigNode, {
+			state: {
+				transientModelConfig: modelConfig
+			},
+			output: {
+				value: [this.modelConfigId],
+				state: _.omit(modelConfigNode.state, ['transientModelConfig'])
+			}
+		});
+
+		// Intervention:
+		wf.updateNode(interventionNode, {
+			state: {
+				interventionPolicy
+			},
+			output: {
+				value: [this.interventionPolicyId],
+				state: interventionNode.state
+			}
+		});
+
+		// Dataset state:
+		wf.updateNode(datasetNode, {
+			state: {
+				datasetId: this.datasetId
+			},
+			output: {
+				value: [this.datasetId]
+			}
+		});
+
+		// Simulates
+		wf.updateNode(simulateBaseNode, {
+			state: {
+				defaultSimulateState,
+				chartSettings: this.outputChartSettings
+			}
+		});
+
+		wf.updateNode(simulateWithInterventionNode, {
+			state: {
+				defaultSimulateState,
+				chartSettings: this.outputChartSettings
+			}
+		});
+
+		// Calibrate
+		wf.updateNode(calibrateNode, {
+			state: {
+				defaultCalibrationState,
+				chartSettings: this.outputChartSettings
+			}
+		});
+
+		// Add Edges:
+		// Model Config
+		wf.addEdge(modelNode.id, modelNode.outputs[0].id, modelConfigNode.id, modelConfigNode.inputs[0].id, [
+			POSITION,
+			POSITION
+		]);
+
+		// Intervention
+		wf.addEdge(modelNode.id, modelNode.outputs[0].id, interventionNode.id, interventionNode.inputs[0].id, [
+			POSITION,
+			POSITION
+		]);
+
+		// Calibrate:
+		wf.addEdge(modelConfigNode.id, modelConfigNode.outputs[0].id, calibrateNode.id, calibrateNode.inputs[0].id, [
+			POSITION,
+			POSITION
+		]);
+
+		wf.addEdge(datasetNode.id, datasetNode.outputs[0].id, calibrateNode.id, calibrateNode.inputs[1].id, [
+			POSITION,
+			POSITION
+		]);
+
+		wf.addEdge(interventionNode.id, interventionNode.outputs[0].id, calibrateNode.id, calibrateNode.inputs[2].id, [
+			POSITION,
+			POSITION
+		]);
```

Comment:
I find it a bit strange that the intervention operator feeds into the calibrate operator - that seems to void the idea of counter-factual? that is you are kind of forcing your base-case to be your what-if case, then you ask what is the diff between the base-case and the what-if case.

Reply:
I will check in the design channel